src/serac/infrastructure/input.cpp

```diff
@@ -89,33 +92,100 @@ void BoundaryConditionInputOptions::defineInputFileSchema(axom::inlet::Table& ta
 
 bool CoefficientInputOptions::isVector() const
 {
-  return std::holds_alternative<CoefficientInputOptions::VecFunc>(func);
+  return vector_function || constant_vector || (!pw_const_vector.empty());
 }
 
-mfem::VectorFunctionCoefficient CoefficientInputOptions::constructVector(const int dim) const
+std::unique_ptr<mfem::VectorCoefficient> CoefficientInputOptions::constructVector(const int dim) const
 {
-  auto vec_func = std::get_if<CoefficientInputOptions::VecFunc>(&func);
-  SLIC_ERROR_IF(!vec_func, "Cannot construct a vector coefficient from a scalar function");
-  return {dim, *vec_func};
+  SLIC_ERROR_IF(!isVector(), "Cannot construct a vector coefficient from scalar input");
+
+  if (vector_function) {
+    return std::make_unique<mfem::VectorFunctionCoefficient>(dim, vector_function);
+  } else if (constant_vector) {
+    return std::make_unique<mfem::VectorConstantCoefficient>(*constant_vector);
+  } else if (!pw_const_vector.empty()) {
+    // Find the maximum mesh attribute
+    auto max_attr_elem = std::max_element(pw_const_vector.begin(), pw_const_vector.end(),
+                                          [](auto a, auto b) { return a.first < b.first; });
+
+    // Create the vector array coefficient. We will use this as an array of piecewise constant scalars
+    auto pw_vec_coeff = std::make_unique<mfem::VectorArrayCoefficient>(max_attr_elem->second.Size());
+
+    // Loop over each spatial dimension
+    for (int i = 0; i < max_attr_elem->second.Size(); ++i) {
+      // Create an mfem vector for the attributes
+      // Note that this vector expects zero indexing
+      mfem::Vector pw_constants(max_attr_elem->first);
+      pw_constants = 0.0;
+
+      for (auto& entry : pw_const_vector) {
+        pw_constants(entry.first - 1) = entry.second[i];
+      }
+
+      // Set the spatial dimension coefficient to a newly constructed scalar piecewise coefficient
+      pw_vec_coeff->Set(i, new mfem::PWConstCoefficient(pw_constants));
+    }
+    return pw_vec_coeff;
+
+  } else {
+    SLIC_ERROR("Trying to build a vector coefficient when no appropriate input data exists.");
```

Comment:
Can we give more information in this error on how to fix it? or should that be higher up in an inlet verifier?

Reply:
Hmmm. I think it could be in a verifier. I need to learn how to use them anyway. I can also make this error message better.